mlrun/model_monitoring/controller.py

```diff
@@ -65,27 +62,23 @@ def __init__(
         All the time values are in seconds.
         The start and stop time are in seconds since the epoch.
         """
+        self.project = project
         self._endpoint = endpoint
         self._application = application
         self._first_request = first_request
-        self._kv_storage = get_v3io_client(
-            endpoint=mlrun.mlconf.v3io_api,
-            # Avoid noisy warning logs before the KV table is created
-            logger=create_logger(name="v3io_client", level="error"),
-        ).kv
-        self._v3io_container = self.V3IO_CONTAINER_FORMAT.format(project=project)
         self._stop = last_updated
         self._step = timedelta_seconds
         self._start = self._get_last_analyzed()
 
     def _get_last_analyzed(self) -> Optional[int]:
-        try:
-            data = self._kv_storage.get(
-                container=self._v3io_container,
-                table_path=self._endpoint,
-                key=self._application,
-            )
-        except HttpResponseError as err:
+        monitoring_schedules = mlrun.model_monitoring.get_store_object(
+            project=self.project
+        )
+        last_analyzed = monitoring_schedules.get_last_analyzed(
+            endpoint_id=self._endpoint,
+            application_name=self._application,
+        )
+        if not last_analyzed:
```

Comment:
Suggestion - Can you maybe raise  form the store object `MLRunNotFoundError` in case there the table don't contains the desired record.

Reply:
https://github.com/mlrun/mlrun/pull/5304/commits/ecae9118c61db11c9f124b552d65402e254a63e9